pkg/agent/httpConfigForCluster.go

```diff
@@ -0,0 +1,112 @@
+package agent
+
+import (
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/log"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	diskcached "k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// configForCluster implements the genericclioptions.RESTClientGetter interface
+// while ensuring that it returns the config it was given, rather
+// than re-creating a config as if the CLI options were passed in as the
+// genericclioptions.ConfigFlags implementation, which strips the bearer token
+// if the host is not https (helpful for the CLI).
+//
+// TODO(mnelson) The better short-term option is to update pinniped-proxy to support TLS
+// even though it's for internal traffic only, as it will be required in many circumstances.
+// https://github.com/kubeapps/kubeapps/issues/2268
+// This implementation can be completely removed once TLS is used by pinniped-proxy.
+type configForCluster struct {
+	config         *rest.Config
+	discoveryBurst int
+	*genericclioptions.ConfigFlags
+}
+
+// ToRESTConfig has the main difference from the original implementation,
+// returning the config as is.
+func (f *configForCluster) ToRESTConfig() (*rest.Config, error) {
+	log.Debug("configForCluster.ToRESTConfig called")
```

Comment:
do you want to keep these debug messages?

Reply:
Yes, I did - they were incredibly helpful for debugging, but then, hmm, they are very specific to debugging this issue. Perhaps I'll remove. Thanks.